### ISSUE TITLE
Review Garvin City Collection in Production

### ISSUE THREAD

**Issue by motropuk** [yalelibrary/YUL-DC#1929]

**Comment by motropuk** [comment]
Reassociation note - there are a number of images that were deemed to be out of scope for this collection. However, they are still described in Findit. But the images for these items were unpublished in Findit, e.g https://findit.library.yale.edu/catalog/digcoll:4215161

When moving metadata for this collection from LB into ASpace, need to decide if these items should still be described in collection, or deaccessioned

**Comment by motropuk** [comment]
Prod ingest notes from Maggie https://app.zenhub.com/workspaces/yul-dc-5e50083561915b11fc9e5850/issues/yalelibrary/yul-dc/1807#issuecomment-1061953275

**Comment by motropuk** [comment]
Visability update complete in Management app. Waiting for collection to reindex in Blacklight https://collections.library.yale.edu/catalog?f%5bproject_identifier_tesi%5d%5b%5d=65&q=&search_field=all_fields

**Comment by motropuk** [comment]
Reindex in Blacklight completed https://collections.library.yale.edu/catalog?f%5bproject_identifier_tesi%5d%5b%5d=65&q=&search_field=all_fields. Got to 14172 objects (diff of 2387 from Management)

**Comment by MaggieZhaoYale** [comment]
@motropuk https://collections.library.yale.edu/catalog?f%5bproject_identifier_tesi%5d%5b%5d=65&q=&search_field=all_fields. The number raised to 14823, 874 private in the management app. 
862 diff from management which is very closed to the number of failed    parents
Here is the list of 234 parents:
https://app.zenhub.com/files/242185115/3e928933-2f39-4424-a643-508362440e13/download
627 parents missing images https://app.zenhub.com/files/242185115/04a0b97c-7d66-46e9-adc6-37dc0eaa863e/download

**Comment by motropuk** [comment]
Summer sent problem parents to Sarah Coe. Sarah is reviewing.

**Comment by motropuk** [comment]
All parents now set to Public. Now showing 15659 in Blacklight https://collections.library.yale.edu/catalog?f%5bproject_identifier_tesi%5d%5b%5d=65&q=&search_field=all_fields. Diff between Blacklight and Management is 900. Most of these are the ones Sarah is now working on

**Comment by motropuk** [comment]
Fixing current issues with missing children or failed parents is going to require extensive work in LB. So we are going to propose to arts that we ingest the problematic images that are currently not displaying or in DCS as new objects via Goobi. This will mean they get new OIDs, but I think they will be fine with this. However, this approach will have hold until the data for this collection has been migrated from LB to ASpace. Once that is done, we can complete the steps on this ticket.

**Comment by motropuk** [comment]
created #1967 for reassociation work. Once #1967 is complete, we can return to his ticket and ingest the problem images via Goobi

**Comment by sshetenhelm** [comment]
Stats as of 2022.08.01:
- 16,559 parent objects in MGMT App
- 15,661 parent objects visible in Blacklight
- Difference between MGMT & Backlight - 898

- 3 parent objects with 0 children
- 851 parent objects with no Aspace URI

**Comment by MaggieZhaoYale** [comment]
@sshetenhelm 822 parents without children or some children have no image (nil with or length)
[https://app.zenhub.com/files/242185115/e2633cf6-32a8-4758-9d81-41cce83b0d5e/download](https://app.zenhub.com/files/242185115/e2633cf6-32a8-4758-9d81-41cce83b0d5e/download)

**Comment by sshetenhelm** [comment]
Thank you so much, @MaggieZhaoYale !

**Comment by sshetenhelm** [comment]
Objects with 1 child failure - 79
Objects with 2 or 3 child failures - 48
Objects with 4 or more child failures - 155
Objects with exactly half child failures - 363
Objects with total failures - 174
Objects with 0 children - 3

Currently investigating objects where exactly half of children failed. Out of 30 objects reviewed so far, 29 are instances of where a record was duplicated in LB and only contains 1 "real" child (child #2 has same caption, title, and filename, just a broken image icon). Best path forward seems to be just to delete the extra children in MGMT for these objects.

**Comment by sshetenhelm** [comment]
Out of 363 objects with half child failures: 
- 4 are actual broken children
- 1 is a hierarchical record with "broken children" actually working parents (will delete)
- 358 duplicate records with no unique images 

- [ ] Delete hierarchical record
- [ ] Delete erroneous children
- [ ] Update list in previous comment to reflect updated numbers

**Comment by sshetenhelm** [comment]
- 50 of 81 single child failures were duplicate records - deleted children
- 31 out of 48 two/three child failures were duplicate records - deleted children
- 5 out of 48 two/three child failures were hierarchical records already represented as parents in DCS - deleted children

Investigating 157 “4+” failures now.

**As of 2022.08.26 10:00 am:
Total MGMT - 16,482
Total Blacklight - 16,105
Discrepancy - 377**

**Comment by sshetenhelm** [comment]
Out of 157 “4+” failures:

- 110 were duplicate or hierarchical records — deleted children
- 1 actually total failure
- 19 need to have children reingested
- 27 need review by Sarah Coe - message sent on 2022.09.02 asking for info 

Moving on to investigating total failures. 

**As of 2022.09.02 -** 
Total MGMT - 16,482
Total Blacklight - 16,212
Discrepancy - 270

**Comment by sshetenhelm** [comment]
Out of 822 parent object failures:

- 553 were the result of child objects that failed because they were either hierarchical records or duplicate records with no files attached in Ladybird. 
- 266 objects have one or more broken children that need to be reingested
- 3 objects have zero children in MGMT but need to be reingested

Total MGMT - 16,482
Total Blacklight - 16,213
Discrepancy - 269 (266 + 3)

Moving on to investigate best way to (hopefully) publish to FindIt and then migrate the "usual" way.

**Comment by sshetenhelm** [comment]
Added "waiting" tag, will publish to Hydra after other projects are finished in LB (e-resources issues and Kissinger permissions).

**Comment by sshetenhelm** [comment]
Have started publishing objects to Hydra. Starting with parents with no children in DCS and parents where all children are broken in DCS.

**Comment by MaggieZhaoYale** [comment]
Downloaded the images and saved them in the pairtree. Re-ingested these children https://collections.library.yale.edu/management/batch_processes/10418
https://collections.library.yale.edu/management/batch_processes/10423 (was failed in the previous batch because the image was jpg)

**Comment by sshetenhelm** [comment]
Thanks Maggie! They look great!

Management - 16,473
Blacklight - 16,244
Discrepancy - 229

**Comment by sshetenhelm** [comment]
@MaggieZhaoYale 
Next list is available for whenever you have time -- these are multiple child ingests for parents

Total Parents - 22
Total Children - 509

Spreadsheet - [https://docs.google.com/spreadsheets/d/12dguZBp8p5Tl_tLxot1BkgFTsFRKYOJVsGH2ayuZUl0/edit?usp=sharing](https://docs.google.com/spreadsheets/d/12dguZBp8p5Tl_tLxot1BkgFTsFRKYOJVsGH2ayuZUl0/edit?usp=sharing )

- [ ] Ingest 29 children from spreadsheet into Management Prod

Please let me know if you have any problems or need more information!

**Comment by MaggieZhaoYale** [comment]
https://collections.library.yale.edu/management/batch_processes/10463 (508 children  and 26 parents)

**Comment by sshetenhelm** [comment]
Thanks @MaggieZhaoYale ! One of the parents failed because I missed some children. I added them to the next spreadsheet: [https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit?usp=sharing](https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit?usp=sharing)

- [ ] Ingest 24 children from 12 parents into MGMT PROD

Management - 16,473
Blacklight -  16,269
Discrepancy - 204

**Comment by sshetenhelm** [comment]
Thanks @MaggieZhaoYale !
Next round: [https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit?usp=sharing](https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit?usp=sharing)

Ingest 528 children from 16 parents in MGMT PROD

**Comment by MaggieZhaoYale** [comment]
https://collections.library.yale.edu/management/batch_processes/10528 Ingested 528 children to DCS prod @sshetenhelm

**Comment by sshetenhelm** [comment]
MGMT PROD - 16,473
Blacklight - 16,297
Difference - 176

Thank you @MaggieZhaoYale ! 
Next round: [https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit#gid=0](https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit#gid=0)

- [ ] Ingest 437 children from 19 parents into MGMT PROD

**Comment by MaggieZhaoYale** [comment]
https://collections.library.yale.edu/management/batch_processes/10541 Ingested 436 children into MGMT prod

**Comment by sshetenhelm** [comment]
- 9 parents that are in Ladybird but were never published to FindIt
- 8 parents with child images that were never properly ingested into Ladybird (no files in LB).
- 56 parents with parent metadata in FindIt, but child images were never published 
--Some images appear to be photos or scans of images from publications (ex. 16857987)
- 73 parents with parent metadata in FindIt, but child images were unpublished/made inactive 
--Many/majority from Texas

Asked Sarah Coe to review and provide any insight on why things may be unpublished or never published. Waiting for reply.

Broken images will need to be ingested via Goobi.

**Comment by sshetenhelm** [comment]
@MaggieZhaoYale when you get back from PTO, could you please ingest these 82 parents and 1591 children into MGMT Prod? 
Spreadsheet: [https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit?usp=sharing](https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit?usp=sharing)

- [ ] ingest 82 parents and 1591 children into MGMT Prod

**Comment by MaggieZhaoYale** [comment]
https://collections.library.yale.edu/management/batch_processes/10934 82 parents in MGMT (please ignore the 3 failed parents. They were successfully ingested after a second try) @sshetenhelm

**Comment by sshetenhelm** [comment]
Thanks Maggie!!!! 

MGMT PROD - 16,473
Blacklight - 16,399
Difference - 74

Getting closer!

**Comment by sshetenhelm** [comment]
Next batch ready! @MaggieZhaoYale , could you please import [these 1,823 from 35 parents](https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit?usp=sharing) into MGMT PROD?

- [ ] Ingest 1,823 children from 35 parents into MGMT PROD

**Comment by MaggieZhaoYale** [comment]
@1823 children from 35 parents in MGMT PROD https://collections.library.yale.edu/management/batch_processes/10948

**Comment by sshetenhelm** [comment]
Thanks Maggie!!!! :) 

MGMT PROD - 16,472 _(another hierarchical-only parent was deleted)_
Blacklight - 16,434
**Difference - 38**

Seeing if anything else can go in this way, or if they all need Goobi.

**Comment by sshetenhelm** [comment]
@MaggieZhaoYale could you please ingest [these 1057 children from 13 parents](https://docs.google.com/spreadsheets/d/1gLXprurJSk8hSN71X7gVPJv3gE1Kz2CZInjX449PSws/edit?usp=sharing) into MGMT prod?

- [ ] ingest 1057 children from 13 parents into MGMT prod

Also, there are three parents that have 0 children in MGMT PROD but have children now available in FindIt. Can you bring over the children for these parents, or do you need me to delete the parent record so you can bring over the entire object? Do you need the child OIDs?

- [ ] Ingest 7 children from FindIt to MGMT PROD for parent 16195742
- [ ] Ingest 15 children for parent 16195324
- [ ] Ingest 12 children for parent 16195719

**Comment by MaggieZhaoYale** [comment]
1057 children were ingested into prod https://collections.library.yale.edu/management/batch_processes/11020

**Comment by sshetenhelm** [comment]
Hi @MaggieZhaoYale 

I have found replacement files for the children of 10 parents. Would you please be able to attach them to their parents? 

The images are in **DCS_MaggieandSummer > Garvin Replacements.** There are folders titled with each parent OID, and each child image is named with the child OID.

If you don't have time or if the objects don't work, please just let me know!

- [ ] Ingest replacement child images for 10 parents

**Comment by MaggieZhaoYale** [comment]
@sshetenhelm Images for 10 parents are in place https://collections.library.yale.edu/management/batch_processes/11281
Woohoo!

**Comment by sshetenhelm** [comment]
THANK YOU MAGGIE!

MGMT Prod - 16,472
Blacklight - 16,461
**Difference - 11**

**Comment by sshetenhelm** [comment]
@MaggieZhaoYale Child image files for 9 more parent objects are available in DCS_MaggieandSummer > Garvin Replacements

Folders are named with parent OIDs, files are named with child OID.

- [ ] Ingest child images from 9 parent objects into MGMT PROD

**Comment by MaggieZhaoYale** [comment]
@sshetenhelm images of 9 parents are in production https://collections.library.yale.edu/management/batch_processes/11301

**Comment by sshetenhelm** [comment]
@MaggieZhaoYale Thank you <3

The child images for the final two parents are available in `DCS_MaggieandSummer > Garvin Replacements`. Folders are named with parent OIDs, files are named with child OID.

- [ ] Ingest child images from 2 parent objects into MGMT PROD

**Comment by MaggieZhaoYale** [comment]
The LAST two parent are in prod: https://collections.library.yale.edu/management/batch_processes/11305 @sshetenhelm

**Comment by sshetenhelm** [comment]
MGMT Prod - 16,472
Blacklight - 16,472
**Difference - 0**

WE DID IT

@MaggieZhaoYale is there a handle file?

**Comment by sshetenhelm** [comment]
Handles updated. Ready to close with as many story points as the universe allows.
